kernel/spawn/src/lib.rs

```diff
@@ -587,8 +588,10 @@ fn task_wrapper_internal<F, A, R>() -> Result<R, task::KillReason>
     // Since the `task_switch()` function was originally invoked from an interrupt handler,
     // interrupts were disabled but never had the chance to be re-enabled
     // because we did not return from the interrupt handler as usual.
-    // Therefore, we need to re-enabled interrupts here to ensure that things continue
-    // to run as normal, with interrupts enabled so we can properly preempt this task.
+    // Therefore, we need to re-enable interrupts and preemption here to ensure that
+    // things continue to run as normal, with interrupts and preemption enabled
+    // such that we can properly preempt this task.
+    drop(recovered_preemption_guard);
     enable_interrupts();
```

Comment:
Do we still need this? Do interrupt handlers automatically disable interrupts when called?

Reply:
yes, and yes.